collections-arrays/CompareByteArraysInCsharp/CompareByteArraysInCsharp/CompareByteArrays.cs

```diff
@@ -0,0 +1,147 @@
+﻿using BenchmarkDotNet.Attributes;
+using System.Collections;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+
+namespace CompareByteArraysInCsharp
+{
+    [MemoryDiagnoser]
+    [Orderer(BenchmarkDotNet.Order.SummaryOrderPolicy.FastestToSlowest)]
+    [RankColumn]
+    public class CompareByteArrays
+    {
+        private readonly Random _rand;
+        private readonly byte[] _byteArray;  
+
+        public CompareByteArrays()
+        {
+            _rand = new Random();
+            _byteArray = GenerateRandomArray(10000000);
+        }
+        public IEnumerable<object[]> SampleByteArray()
+        {
+            yield return new object[] { GenerateRandomArray(10000000), GenerateRandomArray(10000000) };
+            yield return new object[] { _byteArray, _byteArray };
+        }
+
+        [ArgumentsSource(nameof(SampleByteArray))]
+        [Benchmark]
+        public bool CompareUsingSequenceEqual(byte[] firstArray, byte[] secondArray)
+        {
+            return firstArray.SequenceEqual(secondArray);
+        }
+
+        [ArgumentsSource(nameof(SampleByteArray))]
+        [Benchmark]
+        public bool CompareUsingExcept(byte[] firstArray, byte[] secondArray)
+        {
+            var onlyFirst = firstArray.Except(secondArray);
+            var onlySecond = secondArray.Except(firstArray);
+
+            return !onlyFirst.Any() && !onlySecond.Any();
+        }
+
+        [ArgumentsSource(nameof(SampleByteArray))]
+        [Benchmark]
+        public bool CompareUsingForLoop(byte[] firstArray, byte[] secondArray)
+        {
+            if (firstArray.Length != secondArray.Length)
+            {
+                return false;
+            }
+
+            for (int i = 0; i < firstArray.Length; i++)
+            {
+                if (firstArray[i] != secondArray[i])
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        [ArgumentsSource(nameof(SampleByteArray))]
+        [Benchmark]
+        public bool CompareUsingHash(byte[] firstArray, byte[] secondArray) 
+        {
+            using (var hashInstance = SHA256.Create()) 
```

Comment:
I originally avoided adding this comment as this method should be used for the purpose of an equality check, but came back and decided to add it so you would have the information.
You don't need the using and the creation of the hasher, you can just call the static method SHA256.HashData

Reply:
Just realized that there is an error in my comment here - what I meant to say was "this method should NOT be used for the purpose of an equality check"